pkg/xpkg/name_test.go

```diff
@@ -65,6 +68,38 @@ func TestFriendlyID(t *testing.T) {
 			},
 			want: "provider-aws-plusabunchofothernonsensethatisgoingt-123456789123",
 		},
+		"ReplacePeriod": {
+			reason: "All period characters should be replaced with a dash.",
+			args: args{
+				pkg:  "provider.aws-plusabunchofothernonsensethatisgoingtogetslicedoff",
+				hash: "1234.567891234567",
+			},
+			want: "provider-aws-plusabunchofothernonsensethatisgoingt-1234-5678912",
+		},
+		"LeadingDash": {
+			reason: "Leading dash should be replaced with alphanumeric character.",
+			args: args{
+				pkg:  "-provider-aws",
+				hash: "1234567",
+			},
+			want: "Wprovider-aws-1234567",
```

Comment:
> We explicitly use a random alphanumeric character so that replacement behavior is not
predictable.

Can you elaborate on this? These names look a little unusual with the random suffix and prefix. Would we be at risk of a collision if we just trimmed them?

Reply:
@negz good point. While unlikely, we could encounter collision, but the situation would only be marginally improved by this randomization. For instance, `crossplane/provider-foo:sometaghere-v2` and `crossplane/provider-foo:sometaghere-v3` would evaluate the same without randomization (truncate tag at 12 characters, then trim the `-`) and differently with, but `crossplane/provider-foo:sometagherev1` and `crossplane/provider-foo:sometagherev2` would evaluate the same with or without randomization. In addition, in practice, this function is always being called with a digest, not a tag, so we wouldn't encounter this anyway. Unless we think that there is benefit in keeping a name that is >= to 63 characters evaluating to = 63 characters then I think it is much simpler to just strip off `-` as you suggest. Since we are expecting a valid DNS subdomain being passed, do you think we should even check for a leading `-`? A valid DNS subdomain could not lead with it.

Reply:
I'd prefer to trim rather than replace, even if it creates a slightly risk of collisions. If our input could not possibly start with an invalid character in practice it's probably overkill to address that use case.